vignettes/decorate-modules-output.Rmd

```diff
@@ -0,0 +1,389 @@
+---
+title: "Customizing Module Output"
+author: "NEST CoreDev"
+output:
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Customizing Module Output}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+`teal` is a powerful `shiny`-based framework with built-in modules for interactive data analysis. 
+This document outlines the customization options available for modifying the output of `teal` modules. 
+You will learn how to use `teal_transform_module` to modify and enhance the objects created by `teal` modules, 
+enabling you to tailor the outputs to your specific requirements without rewriting the original module code.
+
+Adjusting input data or customizing module outputs in `teal` is accomplished using `transforms` created through 
+`teal_transform_module`. This document will focus on modifying outputs, and for clarity, we will refer to these 
+output-modifying objects as decoration objects or decorators.
+
+Throughout the examples, you will see references to a `data()` object. 
+This reactive object, built with `teal_data()`[https://insightsengineering.github.io/teal.data/latest-tag/articles/teal-data.html],
+holds all the data used `within` the module as well as the generated output objects. 
+`teal_transform_module` operates on this object and can execute custom R code within the `data()` environment using the `within function`. 
+This approach allows you to pass code directly and evaluate it in the `data()` environment.
+
+## Decorators
+
+### Server
+
+The simplest way to create a decorator is to use `teal_transform_module` without specifying a `ui`.
```

Comment:
should be have a `teal_decorator_module`?

Reply:
Unsure, I think using `teal_transform_module` to show that it reuses some logic is OK

Reply:
And if we have `teal_transform_module` (with a verb in the name) then for decorators we should have `teal_decorate_module`

Reply:
But maybe with `teal_decorate_module` it would be easier to document and explain and show the different purpose, even though it's quite similar. I'm puzzled :D